test/cases/fixtures_test.rb

```diff
@@ -0,0 +1,152 @@
+require "cases/helper_cockroachdb"
+
+# Load dependencies from ActiveRecord test suite
+require "cases/helper"
+require "models/account"
+require "models/company"
+require "models/course"
+
+module CockroachDB
+  class FixturesResetPkSequenceTest < ActiveRecord::TestCase
+    self.use_transactional_tests = false
+
+    # Drop and recreate the accounts, companies, and courses tables so they use
+    # primary key sequences. After recreating the tables, load their fixtures.
+    # We'll do this in a before_setup so we get ahead of
+    # ActiveRecord::TestFixtures#before_setup.
+    def before_setup
+      Account.connection.drop_table :accounts, if_exists: true
+      Account.connection.exec_query("CREATE SEQUENCE accounts_id_seq")
+      Account.connection.exec_query("
+        CREATE TABLE accounts (
+          id BIGINT PRIMARY KEY DEFAULT nextval('accounts_id_seq'),
+          firm_id bigint,
+          firm_name character varying,
+          credit_limit integer
+        )
+      ")
+
+      Company.connection.drop_table :companies, if_exists: true
+      Company.connection.exec_query("CREATE SEQUENCE companies_nonstd_seq")
+      Company.connection.exec_query("
+        CREATE TABLE companies (
+          id BIGINT PRIMARY KEY DEFAULT nextval('companies_nonstd_seq'),
+          type character varying,
+          firm_id bigint,
+          firm_name character varying,
+          name character varying,
+          client_of bigint,
+          rating bigint,
+          account_id integer,
+          description character varying
+        )
+      ")
+
+      Course.connection.drop_table :courses, if_exists: true
+      Course.connection.exec_query("CREATE SEQUENCE courses_id_seq")
+      Course.connection.exec_query("
+        CREATE TABLE courses (
+          id INT PRIMARY KEY DEFAULT nextval('courses_id_seq'),
+          name character varying,
+          college_id integer
+        )
+      ")
+
+      self.class.fixtures :accounts
+      self.class.fixtures :companies
+      self.class.fixtures :courses
+    end
+
+    def setup
+      @instances = [Account.new(credit_limit: 50), Company.new(name: "RoR Consulting"), Course.new(name: "Test")]
+      ActiveRecord::FixtureSet.reset_cache # make sure tables get reinitialized
+    end
+
+    # Drop the primary key sequences and bring back the original tables.
+    def teardown
+      Account.connection.drop_table :accounts, if_exists: true
+      Account.connection.exec_query("DROP SEQUENCE IF EXISTS accounts_id_seq")
+      Account.connection.create_table :accounts, force: true do |t|
+        t.references :firm, index: false
+        t.string  :firm_name
+        t.integer :credit_limit
+      end
+
+      Company.connection.drop_table :companies, if_exists: true
+      Company.connection.exec_query("DROP SEQUENCE IF EXISTS companies_nonstd_seq")
+      Company.connection.create_table :companies, force: true do |t|
+        t.string  :type
```

Comment:
In these tests, we're dropping tables, recreating them with primary key sequences, and bringing back the original tables. This _could_ lead to problems because we're bringing back the tables based on definitions that _could_ change in future versions of Rails. I don't think that's likely, but if it happens we can create our own tables with primary key sequences and leave the Rails' ones alone.

Even though this is a little janky, it keeps us pretty closely aligned to the original AR tests.